rhino/src/main/java/org/mozilla/javascript/Context.java

```diff
@@ -174,8 +174,9 @@ public class Context implements Closeable {
      * {@link #VERSION_DEFAULT} or is at least {@link #VERSION_1_6}.
      *
      * @since 1.6 Release 1
+     * @deprecated This flag may be removed later.
      */
-    public static final int FEATURE_E4X = 6;
+    @Deprecated public static final int FEATURE_E4X = 6;
```

Comment:
I think that this is a fine change and we should do it, but I still don't think that we should deprecate this flag until it really does something. IMO, deprecating this flag is like saying that we're going to deprecate the E4X feature and I don't think that we're doing that.

Reply:
I think there is still a misunderstanding here...

> until it really does something

the flag currently does something, it controls, if E4X is enabled, when the plugin is present. So currently, you have to enable the flag **and** ensure, that the plugin is in classpath. I see these options:

### add xml-plugin to rhino-all and keep the flag
If someone uses rhino-all as dependency, he will get the plugin automatically and the flag behaves like before. This solution provides maximum backward compatibility (and this is how it is implemented now, but it is annotated as deprecated)

### remove the flag completely
As the flag is enabled by default (at least for languageVersion >= 1.6), we could move that check to the plugin. So if someone has decided to overrride FEATURE_E4X in his factory, this will be a breaking change, as code will no longer compile.

### something between the 2 solutions
deprecate the flag - with a clear hint, that **disabling** E4X (that is the purpose that the flag currently has IMHO) with the flag will be removed in the future and should be controlled with the plugin mechanism.